multiversx_sdk/core/transaction.py

```diff
@@ -21,7 +22,9 @@ def __init__(self,
                  options: Optional[int] = None,
                  guardian: Optional[str] = None,
                  signature: Optional[bytes] = None,
-                 guardian_signature: Optional[bytes] = None) -> None:
+                 guardian_signature: Optional[bytes] = None,
+                 relayer: Optional[str] = None,
+                 inner_transactions: Optional[Sequence[ITransaction]] = None) -> None:
```

Comment:
I don't know if we should do `Optional[Sequence["Transaction"]]` or leave it as it is. Just thinking out loud.

Reply:
@andreibancioiu

Reply:
Indeed, debatable. 

Being a mere data class, `Optional[Sequence["Transaction"]]` would be good (e.g. a data class should preferably hold only other data classes). On the other hand, having it with the interface, `inner_transactions` can be populated with other objects that satisfy `ITransaction` - however, in practice, that might be a quite exotic case.

For simplicity, we can now use the actual class. In time, if we learn that differently (through usage), we can switch to the interface without a breaking change. The other way around would involve a breaking change.